widgets/src/components/TaskList/TaskListContainer.js

```diff
@@ -75,19 +75,27 @@ class TaskList extends React.Component {
       // dispatch onSelectTask event for the first item on list
       onSelectTask(rows[0]);
 
-      this.setState({
-        loading: false,
-        columns: normalizeColumns(JSON.parse(config.columns), rows[0], options, {
-          openDiagram: this.openDiagram,
-          selectTask: onSelectTask,
-        }),
-        rows,
-        lastPage: taskList.metadata.lastPage === 1,
-        size: taskList.metadata.size,
-        connection: config.knowledgeSource,
-      });
+      if (!taskList.payload) {
+        throw new Error('messages.errors.errorResponse');
+      }
+
+      if (!taskList.payload.length) {
```

Comment:
we can check for the `rows.length` here

Reply:
@gReis89 , sorry I have still used `taskList.payload` to check if list is empty or not. This is because I have moved the normalization of rows after checking its length. I thought this would be code-wise efficient. Let me know if you think otherwise.